### PR TITLE
Item

### PR DESCRIPTION
**Pull request by ovekyc**
Item version 2 in progress.....
please check commit message

**This branch will be merged with master after we complete item version 2 code.**

Before complete version 2 code, we keep master branch as version 1 code for front-end development

### CODE REVIEW

File: src/server/items/controllers.js [hunk 1]
```diff
@@ -41,20 +42,37 @@ export default {
     });
   },
   remove: (req, res, cb) => {
+    const GEOHASH_START_POS = 5;
```
Comment (ovekyc) [inline src/server/items/controllers.js]: Please make a module to deal it or just remove constant

File: src/server/items/controllers.js [hunk 2]
```diff
@@ -41,20 +42,37 @@ export default {
...
             statusCode: 400,
-            message: 'Bad Request, No data'
+            message: 'Item was not found'
           }));
         }
-        return cb(new APIError(err));
+        return cb(new APIError(getErr));
+      }
+      const item = value;
+      const itemTimeStamp = MAX_TIME - Number(new Date(item.createdDate));
```
Comment (ovekyc) [inline src/server/items/controllers.js]: we can make time stamp module

File: src/server/items/controllers.js [hunk 3]
```diff
@@ -41,20 +42,37 @@ export default {
...
+      const ops = [];
+      for (let i = 0; i < DEFAULT_PRECISON; i = i + 1) {
+        const ghSubstr = itemGeohash.substring(0, i + 1);
+        const deletedItemId = `item-2-${ghSubstr}-${itemTimeStamp}-${itemUuid}`;
```
Comment (ovekyc) [inline src/server/items/controllers.js]: Use constant in models.js

File: src/server/items/controllers.js [hunk 4]
```diff
@@ -41,20 +42,37 @@ export default {
...
+        ops.push({type: 'put', key: deletedItemId, value: {ref: key}});
+        for (let j = 0; j < 2; j = j + 1) {
+          const itemIndexingId = `item-${j}-${ghSubstr}-${itemTimeStamp}-${itemUuid}`;
```
Comment (ovekyc) [inline src/server/items/controllers.js]: Please rename the variable

File: src/test/node/items/controllers.js [hunk 5]
```diff
@@ -193,39 +240,96 @@ test('modify an item in database', t => {
...
-      t.end();
+    const addRes = httpMocks.createResponse();
+    ItemController.add(addReq, addRes, () => {
+      const key = addRes._getData().data;
+      let indexingItemsCnt = 0;
+      let itemCnt = 0;
+      let error;
+      let statusCode;
+      testDB.createReadStream({
+        start: '0',
```
Comment (ovekyc) [inline src/test/node/items/controllers.js]: '\x00'

File: src/test/node/items/controllers.js [hunk 6]
```diff
@@ -193,39 +240,96 @@ test('modify an item in database', t => {
...
-    ItemController.remove(req, res, () => {
-      const status = res.statusCode;
-      const message = res._getData().message;
-      t.equal(status, expected.status,
-        'should be same status');
-      t.equal(message, expected.message,
-        'should be same message');
-      t.end();
+    const addRes = httpMocks.createResponse();
+    ItemController.add(addReq, addRes, () => {
```
Comment (ovekyc) [inline src/test/node/items/controllers.js:310]: I think it is not good using add method in test of delete

File: src/test/node/items/controllers.js [hunk 7]
```diff
@@ -193,39 +240,96 @@ test('modify an item in database', t => {
...
+          });
+          const res = httpMocks.createResponse();
+          let undeletedItemCnt = 0;
+          ItemController.remove(req, res, () => {
+            const status = res.statusCode;
+            const message = res._getData().message;
+            t.equal(status, expected.status, 'should be same status');
+            t.equal(message, expected.message, 'should be same message');
+            testDB.createReadStream({
+              start: '0',
```
Comment (ovekyc) [inline src/test/node/items/controllers.js]: \x00

**Comment by ovekyc** [comment]
LGTM

**Comment by festinalente91** [comment]
LG2M

**Comment by hewonjeong** [comment]
LGTM

**Pull request merged**